### PR TITLE
Mr Globtastic's Globtacular Globbing Globs

### PR DESCRIPTION
**Pull request by cjllanwarne**
Good news: this PR reinstates globs and almost all relevant centaur tests.
Bad news: I gave the PR such an awful name that nobody wants to talk about it out loud.

### CODE REVIEW

File: centaur/src/main/resources/standardTestCases/globbingscatter.test [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
 
 files {
   wdl: globbingscatter/globbingscatter.wdl
 }
+
+metadata {
+  workflowName: globbingindex
+  "outputs.globbingscatter.combiner1_result": "globbing is my number 1 best hobby\nglobbing is my number 2 best hobby\nglobbing is my number 3 best hobby\nglobbing is my number 4 best hobby\nglobbing is my number 5 best hobby"
+  "outputs.globbingscatter.combiner2_result": "globbing is my number 1 best hobby\nglobbing is my number 2 best hobby\nglobbing is my number 3 best hobby\nglobbing is my number 4 best hobby\nglobbing is my number 5 best hobby"
+}
```
[the remark this answers is not in the document]
Comment (geoffjentry) [reply centaur/src/main/resources/standardTestCases/globbingscatter.test]: Testing is our middle name

File: centaur/src/main/resources/standardTestCases/lots_of_inputs.test [hunk 2]
```diff
@@ -1,8 +1,17 @@
-ignore: true
+# This test makes sure that:
```
Comment (geoffjentry) [inline centaur/src/main/resources/standardTestCases/lots_of_inputs.test]: This seems like something we should be better about doing on centaur tests

File: wdl/src/main/scala/wdl/WdlExpression.scala [hunk 3]
```diff
@@ -227,7 +227,7 @@ final case class WdlWomExpression(wdlExpression: WdlExpression, from: Option[Sco
 
       override def stderr(params: Seq[Try[WomValue]]): Try[WomFile] = ioFunctionSet.stderr(params)
 
-      override def glob(path: String, pattern: String): Seq[String] = ioFunctionSet.glob(pattern)
+      override def globHelper(pattern: String): Seq[String] = ioFunctionSet.glob(pattern)
```
[the remark this answers is not in the document]
Comment (geoffjentry) [reply wdl/src/main/scala/wdl/WdlExpression.scala]: FWIW I also raised an eyebrow at `Helper` but it's worth noting that I'm usually skeptical of such names so meh

File: centaur/src/main/resources/standardTestCases/lots_of_inputs.test [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
 
 files {
   wdl: lots_of_inputs/lots_of_inputs.wdl
 }
+
+metadata {
+    workflowName: lots_of_inputs
+    status: Succeeded
+    "outputs.lots_of_inputs.out_count": "400"
+    "outputs.lots_of_inputs.nothing_out": "no-op"
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply centaur/src/main/resources/standardTestCases/lots_of_inputs.test]: Perhaps, but it doesn't cost us to check it, and it makes sure the second task runs to completion

File: wdl/src/main/scala/wdl/WdlExpression.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply wdl/src/main/scala/wdl/WdlExpression.scala]: It's to make it clear that this is the end of the IoFunctions layer, and not a WDL engine function, but merely what a `WdlFunction` implementer has to add to enable globbing.
Alternative suggestions welcome...

File: centaur/src/main/resources/standardTestCases/space/space.wdl [hunk 5]
```diff
@@ -20,16 +20,16 @@ task goodbye {
     Array[File] files
     Array[File] files2
     command {
-        echo "Goodbye ${f}"
-        echo "Goodbye ${f2}"
-        echo "Goodbye ${sep = " " files}"
-        echo "Goodbye ${sep = " " files2}"
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply centaur/src/main/resources/standardTestCases/space/space.wdl]: that's the great thing about never checking outputs...!

**Pull request merged**